website/sales/models/order.py

```diff
@@ -0,0 +1,213 @@
+from decimal import Decimal
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.core.validators import MinValueValidator
+from django.db import models
+from django.db.models import (
+    Sum,
+    Value,
+    F,
+)
+from django.db.models.functions import Coalesce
+from django.urls import reverse
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
+from queryable_properties.managers import QueryablePropertiesManager
+from queryable_properties.properties import (
+    RelatedExistenceCheckProperty,
+    AggregateProperty,
+)
+
+from members.models import uuid, Member
+from payments.models import Payable, Payment
+from sales.models.product import ProductListItem
+from sales.models.shift import Shift
+
+
+def default_order_shift():
+    return Shift.objects.filter(active=True).first()
+
+
+class Order(models.Model, Payable):
+
+    objects = QueryablePropertiesManager()
+
+    class Meta:
+        verbose_name = "order"
+        verbose_name_plural = "orders"
+        permissions = [
+            ("custom_prices", "Can use custom prices and discounts in orders"),
+        ]
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    created_at = models.DateTimeField(default=timezone.now)
+
+    # created_by = models.ForeignKey(Member, on_delete=SET_NULL, related_name='orders_created',) # TODO add this field
```

Comment:
Should this still be added?

Reply:
Yes I think this should be desirable.

Reply:
I decided to implement this using LogEntry objects